### PR TITLE
examples/netpkt: add ethercat test example for robot

### PR DESCRIPTION
**Pull request by zhhyu7**
## Summary
add a simple example to test ethercat's sending and receiving.

## Testing
x86_64 device

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The provided PR description **partially meets** the NuttX requirements. 

**Here's what's good:**

* **Summary:** It clearly states the purpose of the PR - adding a test example for EtherCAT.
* **Testing:** It mentions the build host (x86_64) and implies the target is a device (presumably running NuttX). However, more details are needed (see below).

**Here's what needs improvement:**

* **Summary:**
    *  **What functional part?** Be specific about which part of the codebase (e.g., drivers/net/ethercat/) is affected.
    * **How does it work?** Briefly describe the example's logic - what data is sent/received, how it's verified, etc.
    * **Related issues/PRs:**  Link any relevant issues or PRs in the NuttX or NuttX-Apps repositories.

* **Impact:**  Address **ALL** impact points, even if the answer is "NO." Provide brief explanations for "YES" answers:
    * **New feature?** This seems like a "YES" - clarify if it's for users or just for development/testing.
    * **User impact?**  Likely "NO" if it's just a test example, but confirm.
    * **Build impact?**  Does this add any new build dependencies or options? 
    * **Hardware impact?**  Does this example require specific hardware or EtherCAT configurations?
    * **Documentation impact?**  Does this example warrant an update or addition to the documentation?
    * **Security impact?**  Unlikely, but analyze if the example handles any sensitive data.
    * **Compatibility impact?**  Unlikely, but analyze if it introduces any changes to the EtherCAT interface.

* **Testing:**
    * **Target:** Specify the exact architecture (e.g., ARM Cortex-M4), board, and configuration used for testing.
    * **Logs:**  Instead of placeholder comments, include **actual** relevant log snippets from before and after the change. These logs should demonstrate the example's functionality. 

**Recommendation:**

Revise the PR description to address the missing information and provide more specific details. A well-structured and informative PR description increases the chances of quick review and merging.

File: examples/netpkt/netpkt_ethercat.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  if (sockfd == -1)
+    {
+      perror("ERROR: creating socket failed");
+      return -errno;
+    }
+
+  if (argc >= 2)
+    {
+      ifname = argv[1];
```
Comment (cederom) [inline examples/netpkt/netpkt_ethercat.c:68]: assigning variable length input to const char[4] ?

Comment (zhhyu7) [reply examples/netpkt/netpkt_ethercat.c:77]: > assigning variable length input to const char[4] ?

Is it modified like this `int main(int argc, FAR const char *argv[4])`? Done.

Comment (xiaoxiang781216) [reply examples/netpkt/netpkt_ethercat.c:77]: why make this change? all.main should havr the same prototype
int main(int argc, char *argv[]);

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**